packages/react-scripts/config/env.js

```diff
@@ -24,6 +24,7 @@ if (!NODE_ENV) {
 
 // https://github.com/bkeepers/dotenv#what-other-env-files-can-i-use
 var dotenvFiles = [
+    process.env.CUSTOM_ENV_FILE && `${process.env.CUSTOM_ENV_FILE}`, // support custom env file. Will be useful for CI
```

Comment:
Please change the name of the environment variable to `REACT_APP_ENV` and add support for a `.local` env file.